isthmus/src/main/java/io/substrait/isthmus/expression/FunctionConverter.java

```diff
@@ -69,9 +73,51 @@ public FunctionConverter(
       }
     }
 
+    for (var entry : alm.entries()) {
+      String key = entry.getKey();
+      var func = entry.getValue();
+      for (FunctionMappings.Sig sig : calciteOperators.get(key)) {
+        substraitFuncKeyToSqlOperatorMap.put(func.key(), sig.operator());
+      }
+    }
+
     this.signatures = matcherMap;
   }
 
+  protected Optional<SqlOperator> getSqlOperatorFromSubstraitFunc(
+      Expression.ScalarFunctionInvocation expr) {
+    var resolver = getTypeBasedResolver();
+    if (!substraitFuncKeyToSqlOperatorMap.containsKey(expr.declaration().key())) {
```

Comment:
for all the returns below, can we add a verify step that output derivation is consistent between what we determined in substrait plan and what is in calcite plan?

Reply:
I added a checking after `getSqlOperatorFromSubstraitFunc` returned with a SqlOperator.  Check SqlOperator's returntype, by using it's `SqlReturnTypeInference`, and infer the return type from list of operands.  The return type is compared with substrait function's return type.